pkg/app/piped/executor/waitapproval/waitapproval.go

```diff
@@ -161,3 +195,28 @@ func (e *Executor) getMentionedAccounts(event model.NotificationEventType) ([]st
 
 	return notification.FindSlackAccounts(event), nil
 }
+
+func (e *Executor) validateApproverNum(approver string, num int) (string, bool) {
+	if num <= 1 {
+		e.LogPersister.Infof("Got approval from \"%s\"", approver)
+		return approver, true
+	}
+	as, ok := e.MetadataStore.Stage(e.Stage.Id).Get(approversKey)
+	if !ok {
+		e.LogPersister.Infof("Got approval from \"%s\"", approver)
+		e.LogPersister.Infof("Waiting for other approvers...")
+		return approver, false
+	}
+	approvers := strings.Split(as, " ,")
+	for _, a := range approvers {
+		if a == approver {
+			return "", false
+		}
```

Comment:
```suggestion
		if a == approver {
		        e.LogPersister.Infor("Approval from the same user (%s) will not be counted", approver)
			return "", false
		}
```

Reply:
If we add that process, it will generate a lot of logs.

Reply:
![Uploading Screen Shot 2021-11-24 at 18.37.45.png…]()

Reply:
@nghialv 

So, I disagree with this change.

Reply:
If you misunderstand the specification or forget that you have already approved it, you will be stuck with what is going on without the log.

Because of that, I would vote for this suggestion 😃

Reply:
Oh, I see.
I agree with outputting the log. Any ideas to prevent generating a lot of them?
`time.Sleep`?

Reply:
I feel like it doesn't emit any log as long as you don't issue a new command as the same account, isn't it?

Reply:
Yes, it does.

`e.checkApproval` is called every 5 time seconds.
https://github.com/pipe-cd/pipe/pull/2802/files#diff-02a6821aa5f1210155af9610f8bb1b7683ed309d3ad56a533f0505d12ae6ffd6R76-R77

Reply:
> e.checkApproval is called every 5 time seconds

Yes, that function is called periodically but once a command is handled it will be marked as `handled` and be removed from the command list in the next call. So I guess no problem here.
https://github.com/pipe-cd/pipe/blob/master/pkg/app/piped/executor/waitapproval/waitapproval.go#L92-L99

As your log screenshot, I think we are facing a bug around marking a command as handled. So let's ensure that we are not missing this https://github.com/pipe-cd/pipe/blob/master/pkg/app/piped/executor/waitapproval/waitapproval.go#L112 anywhere.

Reply:
I found a solution to fix it.